tests-integration/src/tests/factory.rs

```diff
@@ -754,5 +758,1458 @@ fn test_add_liquidity() {
+            amount: Uint128::from(10_000u128),
+            token_type: euclid::token::TokenType::Native {
+                denom: "eucl".to_string(),
+            },
+        },
+        token_2: TokenWithDenomAndAmount {
+            token: Token::create("nibi".to_string()).unwrap(),
+            amount: Uint128::from(100_000u128),
+            token_type: euclid::token::TokenType::Native {
+                denom: "nibi".to_string(),
+            },
+        },
+    };
+    register_token(
+        &interchain,
+        &factory,
+        pair_info.token_1.to_token_with_denom(),
+    );
+
+    create_pool(&interchain, &factory, pair_info.clone(), BPS_1_PERCENT);
+
+    // adding funds
+    let chain = interchain
+        .get_chain(factory.environment().chain_id().as_str())
+        .unwrap();
+    let mut funds = vec![];
+    for token in pair_info.get_vec_token_info() {
+        faucet(
+            &chain,
+            chain.sender.as_str(),
+            token.amount.u128() + 10,
+            token.token_type.clone(),
+            &mut funds,
+        );
+    }
+
+    add_liquidity(
+        &interchain,
+        &factory,
+        pair_info,
+        BPS_100_PERCENT,
+        None,
+        funds,
+    );
+}
+
+#[test]
+fn test_add_liquidity_with_timeout() {
+    let sender = Addr::unchecked("sender_for_all_chains").into_string();
+    let interchain = MockInterchainEnv::new(vec![("osmosis", &sender), ("nibiru", &sender)]);
+    let router_chain = interchain.get_chain("nibiru").unwrap();
+
+    let router = crate::helpers::chains::setup_router(&router_chain);
+    let factory = crate::helpers::chains::setup_factory(&interchain, "osmosis", "nibiru", &router);
+
+    let pair_info = PairWithDenomAndAmount {
+        token_1: TokenWithDenomAndAmount {
+            token: Token::create("eucl".to_string()).unwrap(),
+            amount: Uint128::from(10_000u128),
+            token_type: euclid::token::TokenType::Native {
+                denom: "eucl".to_string(),
+            },
+        },
+        token_2: TokenWithDenomAndAmount {
+            token: Token::create("nibi".to_string()).unwrap(),
+            amount: Uint128::from(100_000u128),
+            token_type: euclid::token::TokenType::Native {
+                denom: "nibi".to_string(),
+            },
+        },
+    };
+    register_token(
+        &interchain,
+        &factory,
+        pair_info.token_1.to_token_with_denom(),
+    );
+
+    create_pool(&interchain, &factory, pair_info.clone(), BPS_1_PERCENT);
+
+    // adding funds
+    let chain = interchain
+        .get_chain(factory.environment().chain_id().as_str())
+        .unwrap();
+    let mut funds = vec![];
+    for token in pair_info.get_vec_token_info() {
+        faucet(
+            &chain,
+            chain.sender.as_str(),
+            token.amount.u128(),
+            token.token_type.clone(),
+            &mut funds,
+        );
+    }
+
+    add_liquidity(
```

Comment:
In cross-chain test cases we need to set assertions on the sending and receiving chain to make sure everything is working as expected. Errors on the receiving chain won't show here. 
This applies to all cases where were expect the transaction to work.

Reply:
@joemonem  fixed some changes let me know if there should be any other assertions that need to be added